File: Estruturais/Bridge.py
# Interface do dispositivo que pode ser ligado e desligado
class Dispositivo:
    def ligar(self):
        pass

    def desligar(self):
        pass

# Implementação do dispositivo TV
class TV(Dispositivo):
    def ligar(self):
        print("TV ligada")

    def desligar(self):
        print("TV desligada")

# Controle remoto universal (abstração)
class ControleRemoto:
    def __init__(self, dispositivo):
        self.dispositivo = dispositivo

    def ligar(self):
        self.dispositivo.ligar()

    def desligar(self):
        self.dispositivo.desligar()

# Exemplo de uso
tv = TV()
controle = ControleRemoto(tv)
controle.ligar()  # TV ligada
controle.desligar()  # TV desligada

File: Estruturais/Bridge.java
package Estruturais;

// Interface do dispositivo
interface Dispositivo {
    void ligar();
    void desligar();
}

// Implementação do dispositivo TV
class TV implements Dispositivo {
    public void ligar() {
        System.out.println("TV ligada");
    }

    public void desligar() {
        System.out.println("TV desligada");
    }
}

// Controle remoto que usa qualquer dispositivo que implemente a interface Dispositivo
class ControleRemoto {
    private Dispositivo dispositivo;

    public ControleRemoto(Dispositivo dispositivo) {
        this.dispositivo = dispositivo;
    }

    public void ligar() {
        dispositivo.ligar();
    }

    public void desligar() {
        dispositivo.desligar();
    }
}

public class Bridge {
    public static void main(String[] args) {
        Dispositivo tv = new TV();
        ControleRemoto controle = new ControleRemoto(tv);
        controle.ligar();  // TV ligada
        controle.desligar();  // TV desligada
    }
}

